src/crab/ebpf_domain.cpp

```diff
@@ -2374,7 +2374,8 @@ void ebpf_domain_t::lshr(const Reg& dst_reg, int imm, int finite_width) {
             }
         }
         m_inv.set(dst.uvalue, interval_t{lb_n, ub_n});
-        if (ub_n.narrow<int64_t>() >= lb_n.narrow<int64_t>()) {
+        if ((ub_n.fits<int64_t>() && lb_n.fits<int64_t>()) &&
```

Comment:
If finite_width is 64 and imm is 0, then [lb,ub] can be something that would be (say) [-2,-2] as a signed number.
That is legal and should not be havoced.  So I think this should use `fits_cast_to` not `fits`.